### PR TITLE
Create prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth Rule

### PR DESCRIPTION
**Pull request by rayane-djouah**
https://github.com/Expensify/App/issues/30528

Test:
1. `npm install git+https://github.com/Expensify/eslint-config-expensify.git#1d72b89949db775b76e7f2a39707720d24d07d1c`
2. `npm run lint`

Screenshot 2024-09-15 at 4 15 41 PM

### CODE REVIEW

File: eslint-plugin-expensify/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        ) {
+          return;
+        }
+        context.report({
+          node,
+          message:
+            CONST.MESSAGE
+              .PREFER_SHOULD_USE_NARROW_LAYOUT_INSTEAD_OF_IS_SMALL_SCREEN_WIDTH,
+        });
+      },
```
Comment (getusha) [inline eslint-plugin-expensify/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth.js:95]: I think this is redundant [this](https://github.com/Expensify/eslint-config-expensify/pull/116/files#diff-cf9fee05c9206f3ecb6a39a5793f1887f3d69500ff7703d84589226adf120522R48) handles this case

**Comment by getusha** [comment]
I'll review this, first thing tomorrow. thanks for the bump

**Comment by getusha** [comment]
@rayane-djouah tests failing, do you think this is related?

Screenshot 2024-10-03 at 10 08 38 in the morning

File: eslint-plugin-expensify/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                CONST.MESSAGE
+                  .PREFER_SHOULD_USE_NARROW_LAYOUT_INSTEAD_OF_IS_SMALL_SCREEN_WIDTH,
+            });
+          });
+        }
+
+        // Check for 'const var = useResponsiveLayout();' and use of this var
+        const variableName = node.id.name;
+        const variableUsages = _.filter(
+          context.getScope().references,
```
Comment (getusha) [inline eslint-plugin-expensify/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth.js:43]:
```suggestion
          sourceCode.getScope().references,
```

can we do this? there is a warning

Screenshot 2024-10-03 at 11 18 25 in the morning

**Review by getusha — APPROVED** [review]
Screenshot 2024-10-08 at 3 29 52 in the afternoon

Screenshot 2024-10-08 at 3 40 51 in the afternoon

**Review by mountiny — APPROVED** [review]
Thanks!

**Pull request merged**